app/address/[address]/layout.tsx

```diff
@@ -275,20 +280,32 @@ function AccountHeader({
         let token: { logoURI?: string; name?: string } = {};
         let unverified = false;
 
-        const metadataExtension = mintInfo?.extensions?.find(({ extension }: { extension: string }) => extension === 'tokenMetadata');
-        const metadataPointerExtension = mintInfo?.extensions?.find(({ extension }: { extension: string }) => extension === 'metadataPointer');
-
+        const metadataExtension = mintInfo?.extensions?.find(
+            ({ extension }: { extension: string }) => extension === 'tokenMetadata'
+        );
+        const metadataPointerExtension = mintInfo?.extensions?.find(
+            ({ extension }: { extension: string }) => extension === 'metadataPointer'
+        );
 
         if (metadataPointerExtension && metadataExtension) {
             const tokenMetadata = create(metadataExtension.state, TokenMetadata);
             const { metadataAddress } = create(metadataPointerExtension.state, MetadataPointer);
 
+            (async () => {
+                // Avoid re-renders once tokenData has been set
+                if (!tokenData.name && !tokenData.image) {
+                    const response = await fetch(tokenMetadata.uri);
+                    setTokenData(await response.json());
+                }
+            })();
```

Comment:
instead of using the iife here i think a more canonical React hook would make sure the state works as expected:

```
import React, { useEffect, useState } from 'react';

const MyComponent: React.FC<{ tokenMetadata: { uri: string } }> = ({ tokenMetadata }) => {
    const [tokenData, setTokenData] = useState<{ name?: string, image?: string }>({});

    useEffect(() => {
        const fetchTokenData = async () => {
            if (!tokenData.name && !tokenData.image) {
                const response = await fetch(tokenMetadata.uri);
                const data = await response.json();
                setTokenData(data);
            }
        };

        fetchTokenData();
    }, [tokenMetadata.uri, tokenData.name, tokenData.image]);

    return (
        <div>
            {/* Render token data here */}
            <p>Name: {tokenData.name}</p>
            <img src={tokenData.image} alt={tokenData.name} />
        </div>
    );
};
export default MyComponent;
```